interp/type.go

```diff
@@ -166,56 +166,65 @@ func nodeType(interp *Interpreter, sc *scope, n *node) (*itype, error) {
 
 	case arrayType:
 		t.cat = arrayT
-		if len(n.child) > 1 {
-			v := n.child[0].rval
-			switch {
-			case v.IsValid():
-				// constant size
-				if isConstantValue(v.Type()) {
-					c := v.Interface().(constant.Value)
-					t.size = constToInt(c)
-				} else {
-					t.size = int(v.Int())
+		c0 := n.child[0]
+		if len(n.child) == 1 {
+			// Array size is not defined.
+			if t.val, err = nodeType(interp, sc, c0); err != nil {
```

Comment:
This change has a side-effect: previously, we would do `if t.val, err = nodeType(interp, sc, n.child[0])` even when `len(n.child) == 0`. So I suppose if that happened for some unforeseen reason, we would get a panic and we would at least know we made a mistake somewhere.
With the new code, the case `len(n.child) == 0` is completely skipped. I don't know whether we should be super defensive and add an explicit panic or something in case that happens. I leave that to your appreciation.

Reply:
`len(n.child[0])` means here an array declaration with no type, which I think is not possible in this context, and would have been catch earlier by the Go parser.